sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -407,7 +407,7 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
         val withWindow = withDistinct.optionalMap(windows)(withWindows)
 
         // Hint
-        withWindow.optionalMap(hint)(withHints)
+        hints.asScala.foldRight(withWindow)(withHints)
```

Comment:
so that `select /*+ hint1() /* /*+ hint2() */`produces `Hint1(Hint2(plan))` and not `Hint2(Hint1(plan))`. `withHints` adds a Hint on top so the last one folded is the top most.